app/scripts/modules/core/pipeline/config/preconditions/selector/preconditionSelector.directive.js

```diff
@@ -60,6 +60,16 @@ module.exports = angular.module('spinnaker.core.pipeline.config.preconditions.se
 
       let accountFilter = (cluster) => cluster.account === $scope.precondition.context.credentials;
       $scope.regions = appListExtractorService.getRegions([$scope.application], accountFilter);
+
+      //Setting cloudProvider when account is updated
+      let providerFilter = (accounts) => accounts.name === $scope.precondition.context.credentials;
+      var accountsArray = $scope.accounts;
+      if(accountsArray !== null) {
+        var account = accountsArray.find(providerFilter);
```

Comment:
1). With `var` (as opposed to `let`), `account` will be defined outside the `if` block. Overall, probably better to use `let`, but it's not breaking anything here.
2). If `.find` doesn't find a match, it will return `undefined`, not null, which will cause an error on line 70.